apps/prairielearn/src/pages/shared/syncHelpers.js

```diff
@@ -121,96 +114,81 @@ function logProgressOutput(output, job, printedInfos, prefix) {
-            job.info('Successfully tagged image');
-
-            const repositoryName = repository.getRepository();
-            job.info(`Ensuring repository exists: ${repositoryName}`);
-            confirmOrCreateECRRepo(repositoryName, job, (err) => {
-              if (ERR(err, callback)) return;
-              job.info('Successfully ensured repository exists');
-
-              // Create a new docker image instance with the new registry name
-              // localImage isn't specific enough to the ECR repo
-              const pushImageName = repository.getCombined();
-              const pushImage = new Docker.Image(docker.modem, pushImageName);
-
-              job.info(`Pushing image: ${repository.getCombined()}`);
-              pushImage.push(
-                {
-                  // @ts-expect-error: We seem to be missing a `serveraddress` property,
-                  // but it works fine without it?
-                  authconfig: dockerAuth,
-                },
-                (err, stream) => {
-                  if (ERR(err, callback)) return;
-                  if (!stream) throw new Error('Missing stream from push()');
-
-                  const printedInfos = new Set();
-                  docker.modem.followProgress(
-                    stream,
-                    (err) => {
-                      if (ERR(err, callback)) return;
-                      job.info('Push complete');
-                      callback(null);
-                    },
-                    (output) => {
-                      logProgressOutput(output, job, printedInfos, 'Push progress: ');
-                    },
-                  );
-                },
-              );
-            });
-          });
-        });
+        if (err) reject(err);
+        resolve(null);
       },
       (output) => {
         logProgressOutput(output, job, printedInfos, 'Pull progress: ');
       },
     );
   });
+
+  job.info('Pull complete');
+
+  // Find the image we just downloaded
+  const downloadedImage = repository.getCombined(true);
+  const localImage = docker.getImage(downloadedImage);
+
+  // Tag the image to add the new registry
+  repository.setRegistry(cacheImageRegistry);
+
+  const options = {
+    repo: repository.getCombined(),
+  };
+  job.info(`Tagging image: ${options.repo}`);
+  await localImage.tag(options);
+  job.info('Successfully tagged image');
+
+  const repositoryName = repository.getRepository();
+  job.info(`Ensuring repository exists: ${repositoryName}`);
+  await ensureECRRepo(repositoryName, job);
+  job.info('Successfully ensured repository exists');
+
+  // Create a new docker image instance with the new registry name
+  // localImage isn't specific enough to the ECR repo
+  const pushImageName = repository.getCombined();
+  const pushImage = docker.getImage(pushImageName);
+
+  job.info(`Pushing image: ${repository.getCombined()}`);
+  // @ts-expect-error -- https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69385
```

Comment:
I'm inclined to just live with the `ts-expect-error` here and below until the `DefinitelyTyped` PR lands? That might not be for a while; they aren't always super speedy to approve things.

Reply:
Agreed that we should not wait.